lib/rautomation/adapter/win_32/table.rb

```diff
@@ -9,11 +9,24 @@ class Table < Control
         DEFAULT_LOCATORS = {:class => /syslistview/i}
 
         def select(row)
-          Functions.select_table_row(Window.oleacc_module_handle, Functions.control_hwnd(@window.hwnd, @locators), row)
+          if Platform.is_x86?
```

Comment:
Instead of having these if statements everywhere, I would do something like this instead:
1) In `functions.rb`:

```ruby
attach_function :_get_table_row_state_x86, :get_table_row_state,
                        [:long, :long, :long], :long
attach_function :_get_table_row_state_x64, :get_table_row_state,
                        [:long_long, :long, :long], :long

def get_table_row(oleacc_module_handle, control_hwnd, row)
  send("_get_table_row_state_#{Platform.architecture}", oleacc_module_handle, control_hwnd, row)
end
```

2) In other places (`Table#select` for example):
```ruby
def select(row)
  handle = Window.oleacc_module_handle
  control_hwnd = Functions.control_hwnd(@window.hwnd, @locators)
  state = Functions.get_table_row_state(handle, control_hwnd, row)
  state & Constants::STATE_SYSTEM_SELECTED != 0
end
```

This keeps the messy logic of handling different platforms only in the low-level code (`functions.rb`) and all higher level code does not have (almost) any visible traces of different platform supports. This also means that if in the future there should be a new platform introduces (it is highly unlikely, but for example Apple's M1/M2) then only `functions.rb` needs to be changed in the ideal world and every higher-level code should not be affected.

Reply:
Great suggestion, I wasn't happy with the previous implementation either but wasn't sure on how to make it better. Edited the platform specific function hooks to be called using this method